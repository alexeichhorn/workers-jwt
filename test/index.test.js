const fs = require('fs');
const getGCPJWT = require('../index.js');

describe('gcp-jwt', () => {
  const WebCrypto = require('node-webcrypto-ossl');
  const crypto = new WebCrypto();

  global.Date.now = jest.fn(() => 1530518207007);

  test('getGCPJWT', async () => {
    // XXX Don't worry - this service account was deleted (i.e. can't be abused).
    const s = fs.readFileSync('./assets/service_account.json', {
      encoding: 'utf8',
    });
    const token = await getGCPJWT(s, 'https://baba', crypto);
    expect(token).toMatchSnapshot();
  });
});
