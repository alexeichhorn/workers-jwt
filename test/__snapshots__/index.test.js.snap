// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gcp-jwt getGCPJWT 1`] = `"eyJhbGciOiJSUzI1NiIsImtpZCI6IjY4OTVlZDUyN2U0Mjc4NjEyYzY4M2UzMWVlMjBjMTBlZmNlMTQ4NTEiLCJ0eXAiOiJKV1QifQ.eyJhdWQiOiJodHRwczovL2JhYmEiLCJpc3MiOiJjZnctcmVzdC1hcGktdGVzdDJAYXhhbXBsZS1wcm9qZWN0LmlhbS5nc2VydmljZWFjY291bnQuY29tIiwic3ViIjoiY2Z3LXJlc3QtYXBpLXRlc3QyQGF4YW1wbGUtcHJvamVjdC5pYW0uZ3NlcnZpY2VhY2NvdW50LmNvbSIsImlhdCI6MTUzMDUxODIwNywiZXhwIjoxNTMwNTIxODA3fQ.imaBlJI2H-CEnKLSSpOQuZ7EkOrCciV8ib5CyL0R4rqjBYsplyfXUw-Nf90ZxBxjcG1_OCdYJ0y7zX4X453tRqGdebKu8qx1-Lfya4f6wj4gxiUbJUoEy7-BURfR5LoSlcaYAOuJxzJyl1l7HnipPaCp-zS2ln9VuFWPCz3xjObsQHIKB14dk3GhAw2Nsk27D8-h27Iiygi3B8L-kB2aoUJZb93ve9qZPZ58lzngZBsQK3PXJX_wp8PLgyc1d5XUCmtlUEYRCrOGX94ABkuNZSFsfAIgsW3o1MU15e2_k5X_zeMzpJYD3K7bXu9EwHn3-ofedxuebiCobNBKnJ-WTw"`;
